### PR TITLE
Lifestudy

### PR DESCRIPTION
**Pull request by radicalray**
Finished the lifestudy at a stable point. Merging into dev.

Added on top of Dennis Lu's work:

 - Added support to auto-increment last life study chapter
 - @davidhtien Added celery to regularly calculate and auto-assign life study based on attendance
 - Minor CSS fixes to re-adjust alignments and look cleaner

Good job team! Let's get this merged!

@ricefield @lifedispenser @ludennis

### CODE REVIEW

**Comment by radicalray** [comment]
:+1:  Good job @davidhtien for pushing it to the end!

**Comment by radicalray** [comment]
@ricefield @lifedispenser?

**Comment by ricefield** [comment]
Seems good. I haven't had a chance to do code review yet though. Will try to get to it this week. Are you in a rush to merge?

**Comment by radicalray** [comment]
Yea, would like to merge soon (a lot of code) and work on incremental changes in the future.

**Comment by ricefield** [comment]
hey @radicalray, it mostly looks good to me, I gave it a quick look-over last night. 

I have some misgivings about the usage of jquery to bind to API calls, especially it's inclusion inside the base template. Is this something that has been discussed prior and we are adopting project-wide? Either way, I'm not a fan of this approach and I think we need to have a consistent approach to the front end dev, esp with the usage of JS. I'd also like to get @lifedispenser 's opinion on this.

As far as merging goes for the short-term, if you guys are not willing to make changes to the front-end, you may as well merge it in now and we can always refactor later once we figure out our frontend strategy. If @lifedispenser signs off on this, we're good to go.

**Comment by radicalray** [comment]
@ricefield I didn't put too much thought in the js inclusion. We can refactor in the future, it's a short snippet more so to test out our restful API than for setting a standard. It's included in the base template for lifestudy so the hope is other life study can consume the API automatically without needing a separate import.

File: ap/templates/base.html [hunk 1]
```diff
@@ -27,6 +27,7 @@
       <script src="{% static "js/jquery-1.11.1.min.js" %}"></script>
       <script src="{% static "js/modernizr-2.6.2-respond-1.1.0.min.js" %}"></script>
       <script src="{% static "js/offcanvas.js" %}"></script>
+      <script src="{% static "js/django-jquery-init.js" %}"></script>
```
Comment (ricefield) [inline ap/templates/base.html]: did you mean to put this in a LS-only base template? this is the base template for the entire project @radicalray

Comment (radicalray) [inline ap/templates/base.html]: No, django-jquery-init.js is a fix for a super annoying django admin bug where the 2 versions of jquery (default django admin jquery vs. Our admin skin jquery) conflict with each other and screw up (adding new models) functionality completely.

**Comment by radicalray** [comment]
@ricefield @lifedispenser @davidhtien Let's get this merged?

**Comment by ricefield** [comment]
agreed. @lifedispenser can we get a quick code review from you?

**Comment by radicalray** [comment]
Ok, we are merging this before the term starts. Will work on improvements incrementally from here.

**Pull request merged**